packages/lodestar/src/eth1/eth1JsonRpcClient.ts

```diff
@@ -37,13 +40,31 @@ export class Eth1JsonRpcClient {
   /**
    * Fetches an arbitrary array of block numbers in batch
    */
-  async getBlocksByNumber(blockNumbers: number[], signal?: AbortSignal): Promise<Eth1Block[]> {
+  async getBlocksByNumber(fromBlock: number, toBlock: number, signal?: AbortSignal): Promise<Eth1Block[]> {
     const method = "eth_getBlockByNumber";
-    const blocksRaw = await this.rpc.fetchBatch<IEthJsonRpcTypes[typeof method]>(
-      blockNumbers.map((blockNumber) => ({method, params: [toHex(blockNumber), false]})),
-      signal
+    const blocksRawArr = await retry(
+      (attempt) => {
+        // Large batch requests can return with code 200 but truncated, with broken JSON
+        // This retry will split a given block range into smaller ranges exponentially
+        // The underlying http client should handle network errors and retry
+        const chunkCount = 2 ** (attempt - 1);
+        const blockRanges = chunkifyInclusiveRange(fromBlock, toBlock, chunkCount);
+        return Promise.all(
+          blockRanges.map(([from, to]) =>
+            this.rpc.fetchBatch<IEthJsonRpcTypes[typeof method]>(
+              linspace(from, to).map((blockNumber) => ({method, params: [toHex(blockNumber), false]})),
+              signal
```

Comment:
if signal is activated, won't rpc client throw and this will retry?

Reply:
It will only retry if the error is an instance of `ErrorParseJson`